compiler/rustc_symbol_mangling/src/v0.rs

```diff
@@ -479,7 +479,7 @@ impl<'tcx> Printer<'tcx> for &mut SymbolMangler<'tcx> {
                 })?;
             }
 
-            ty::Dynamic(predicates, r) => {
+            ty::Dynamic(predicates, r, _repr) => {
```

Comment:
We need to mangle dyn and dyn* separately, if you don't want to fix then can you leave a FIXME?

Reply:
I did something here, but I'm not sure it's legal. I assume we also need to adjust demangling too?